pkg/rhos/securitygroups.go

```diff
@@ -0,0 +1,270 @@
+		}
+	}
+
+	pager := servers.List(computeClient, servers.ListOpts{Name: c.InfraID})
+	err = pager.EachPage(func(page pagination.Page) (bool, error) {
+		serverList, err := servers.ExtractServers(page)
+		if err != nil {
+			return false, errors.WithMessage(err, "getting the server List failed")
+		}
+		for i := range serverList {
+			err := secgroups.AddServer(computeClient, serverList[i].ID, groupName).ExtractErr()
+			if err != nil {
+				return false, errors.WithMessage(err, "failed to add the security group to the server")
+			}
+		}
+
+		return true, nil
+	})
+
+	if err != nil {
+		return errors.WithMessagef(err, "failed to open ports")
+	}
+
+	return nil
+}
+
+func (c *CloudInfo) removeInternalFirewallRules(infraID string,
+	computeClient *gophercloud.ServiceClient) error {
+	groupName := infraID + internalSecurityGroupSuffix
+
+	pager := servers.List(computeClient, servers.ListOpts{Name: c.InfraID})
+
+	err := pager.EachPage(func(page pagination.Page) (bool, error) {
+		serverList, err := servers.ExtractServers(page)
+		if err != nil {
+			return false, errors.WithMessage(err, "getting the server List failed")
+		}
+		for i := range serverList {
+			_ = secgroups.RemoveServer(computeClient, serverList[i].ID, groupName).ExtractErr()
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return errors.WithMessage(err, "failed to removed security group from servers")
+	}
+
+	return nil
+}
+
+func (c *CloudInfo) createGWSecurityGroup(ports []api.PortSpec, groupName string, computeClient *gophercloud.ServiceClient,
+	networkClient *gophercloud.ServiceClient) error {
+	isFound, err := checkIfSecurityGroupPresent(groupName, computeClient)
+
+	if isFound {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	opts := secgroups.CreateOpts{
+		Name:        groupName,
+		Description: "Submariner Gateway",
+	}
+
+	group, err := secgroups.Create(computeClient, opts).Extract()
+	if err != nil {
+		return errors.WithMessage(err, "failed to create g/w security group")
+	}
+
+	for _, port := range ports {
+		err = c.createSGRule(group.ID, "", allNetworkCIDR, port.Port, port.Protocol, networkClient)
+		if err != nil {
+			return errors.WithMessagef(err, "creating security group rule failed")
+		}
+	}
+
+	return nil
+}
+
+func checkIfSecurityGroupPresent(groupName string, computeClient *gophercloud.ServiceClient) (bool, error) {
+	pager := secgroups.List(computeClient)
+	var isFound bool
+
+	err := pager.EachPage(func(page pagination.Page) (bool, error) {
+		serverList, err := secgroups.ExtractSecurityGroups(page)
+
+		for _, s := range serverList {
+			if s.Name == groupName {
+				isFound = true
+			}
+		}
+		return true, errors.WithMessagef(err, "failed to extract the security group %q from results", groupName)
+	})
+
+	if isFound {
+		return true, nil
+	}
+
+	if err != nil {
+		return false, errors.WithMessagef(err, "error getting the security group : %q", groupName)
+	}
+
+	return false, nil
```

Comment:
I think this can be simplified to:

```suggestion
	return isFound, errors.WithMessagef(err, "error getting the security group : %q", groupName)
```